suite-native/module-send/src/components/SendOutputsForm.tsx

```diff
@@ -0,0 +1,149 @@
+import { Form, useForm } from '@suite-native/forms';
+import { AccountKey, FormState } from '@suite-common/wallet-types';
+import { useAsyncDebounce } from '@trezor/react-utils';
+import {
+    AccountsRootState,
+    FeesRootState,
+    SendRootState,
+    selectAccountByKey,
+    selectNetworkFeeInfo,
+    selectSendFormDraftByAccountKey,
+    sendFormActions,
+} from '@suite-common/wallet-core';
+import {
+    SendStackParamList,
+    SendStackRoutes,
+    StackNavigationProps,
+} from '@suite-native/navigation';
+import { useToast } from '@suite-native/toasts';
+
+import { onDeviceTransactionReviewThunk } from '../sendFormThunks';
+import { SendOutputsFormValues, sendOutputsFormValidationSchema } from '../sendOutputsFormSchema';
+import { SendOutputFields } from './SendOutputFields';
+
+type SendFormProps = {
+    accountKey: AccountKey;
+};
+
+// TODO: this data structure will be revisited in a follow up PR
+const constructFormDraft = ({ outputs }: SendOutputsFormValues): FormState => ({
+    outputs:
+        outputs?.map(({ address, amount }) => ({
+            type: 'payment',
+            address,
+            amount,
+            token: null,
+            fiat: '0',
+            currency: { label: 'usd', value: '1000' },
+        })) ?? [],
+    isCoinControlEnabled: false,
+    hasCoinControlBeenOpened: false,
+    selectedUtxos: [],
+    feeLimit: '0',
+    feePerUnit: '0',
+    options: [],
+});
+
+export const SendOutputsForm = ({ accountKey }: SendFormProps) => {
+    const dispatch = useDispatch();
+    const debounce = useAsyncDebounce();
+    const { showToast } = useToast();
+    const navigation =
+        useNavigation<StackNavigationProps<SendStackParamList, SendStackRoutes.SendReview>>();
+
+    const account = useSelector((state: AccountsRootState) =>
+        selectAccountByKey(state, accountKey),
+    );
+    const networkFeeInfo = useSelector((state: FeesRootState) =>
+        selectNetworkFeeInfo(state, account?.symbol),
+    );
+    const sendFormDraft = useSelector((state: SendRootState) =>
+        selectSendFormDraftByAccountKey(state, accountKey),
+    );
+
+    const form = useForm<SendOutputsFormValues>({
+        validation: sendOutputsFormValidationSchema,
+        context: {
+            networkFeeInfo,
+            networkSymbol: account?.symbol,
+            availableAccountBalance: account?.availableBalance,
+        },
+        defaultValues: {
+            outputs: [
+                {
+                    address: 'bcrt1q7r9yvcdgcl6wmtta58yxf29a8kc96jkyyk8fsw',
+                    amount: sendFormDraft?.outputs[0]?.amount,
+                },
+            ],
+        },
+    });
+
+    const {
+        handleSubmit,
+        control,
+        getValues,
+        formState: { isValid, isSubmitting },
+    } = form;
+    const watchedFormValues = useWatch({ control });
+
+    const storeFormDraftIfValid = useCallback(async () => {
+        await dispatch(
```

Comment:
good catch. Fixed by: 3b8cbb289f. I had to change typing of `useAsyncDebounce` to allow to send simple synchronous function as a argument.

Reply:
I woul also rename the hook to `useDebounce` as it doesn't only accept async functions now. Or maybe separate into 2 different functions. However I myself prefer the first option.

Reply:
I don't agree. The name `useAsyncDebounce` is telling you, that it makes the call "asynchronously debounce" rather than it accepts only the async functions.

Reply:
I think that's not correct. The async means that the debounce returns a promise. Since it doesn't always return promise anymore, it's not correct and is just redundant. I doubt that it means "asynchronously debounce", since debounce is always like that, that's why it's called `debounce` and not `asynchronous debounce` 😄 I went a little bit far with the metaphor here, but had to prove a point because I think we're looking at this differently. 

Since this is pretty easy to rename (pretty quick if you use the rename symbol in VS code), can we do that to keep this simple and to the point? 

If this was a complicated refactor, it's obviously not too important and could be ignored, but I think we spent more time on this already than if we would just rename it. 

Happy to discuss more if you want though.

Reply:
fixed 96cbb1d01abbc546dc377addafef8a1deb874148